File: About_AI/nodejs/call_antiporn.js
const Stream = require('stream');
const util = require('util');

async function doRequest(event) {
    // get an image from your origin server or somewhere else
    let resp = await event.fetch('http://www.example.com/example.jpg')

    // call duedge ai
    let result = await event.ai.Censor.antiPorn(resp.body);

    if (result['error_code'] > 0) {
        //some errors occured
        event.console.log('error_message:' + result['error_message']);
        return new Response(JSON.stringify(result), {status: 200});
    }

    event.console.log(util.format('The type of this image\'s content is %s', result['conclusion']));

    for (let i = 0;i < result['result'].length;i ++) {
        data = result['result'][i];
        event.console.log(util.format('The probability that the type of this image\'s content is %s: %f' , data['class_name'], data['probability']))
    }

    return new Response(JSON.stringify(result), {status: 200});
}

exports.handler = doRequest;

File: About_AI/python/call_antiporn.py
import json
async def handler(event):
    # get a image from your origin server or internet
    resp = await event.fetch(url='https://www.example.com/example.jpg')

    # call duedge_ai
    result = await event.ai.censor.antiporn(resp.body)

    if result['error_code'] > 0:
        #some errors occured
        event.console.log('error message: %s' % result['error_message'])
        return Response(body=json.dumps(result, ensure_ascii=False), status=200)

    event.console.log('The type of this image\'s content is %s' % result['conclusion'])

    for data in result['result']:
        event.console.log('The probability that the type of this image\'s content is %s: %f' % (data['class_name'], data['probability']))


    return Response(body=json.dumps(result, ensure_ascii=False), status=200)
